workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CreateLicense/index.tsx

```diff
@@ -3,39 +3,186 @@
+	subscription: string;
+};
+
+type ProductCardProps = {
+	licenseKeyData: {[key: string]: string};
+	product: {
+		attachments: [];
+		name: {
+			en_US: string;
+		};
+		productSpecifications: [];
+		skus: {
+			price: number;
+			sku: string;
+			skuOptions: [];
+		}[];
+	};
+	productCreatorAccount: {logoURL: undefined; name: string};
+	userAccount: {[key: string]: string};
+};
+
+type StepsInformationProps = {
+	backStep: string;
+	component: JSX.Element;
+	nextStep: string;
+	stepTitle: string;
+	title: string;
+};
+
+type StepsInformation = {
+	[StepCreateLicense.LICENSE_KEY_DETAILS]: StepsInformationProps;
+	[StepCreateLicense.SUBSCRIPTION]: StepsInformationProps;
+};
+
 const CreateLicense = () => {
 	const [step, setStep] = useState<string>(StepCreateLicense.SUBSCRIPTION);
 
-	const StepsInformation = {
+	const {setValue, watch} = useForm<CreateLicenseForm>({
+		defaultValues: {
+			subscription: undefined,
+		},
+	});
+
+	const {subscription} = watch();
+
+	const stepsInformation: StepsInformation = {
 		[StepCreateLicense.SUBSCRIPTION]: {
 			backStep: StepCreateLicense.SUBSCRIPTION,
-			component: <div>Subscription</div>,
+			component: (
+				<SelectSubscription
+					onSelectSubscription={(subscription: string) => {
+						setValue('subscription', subscription);
+					}}
+					selectedSubscriptionValue={subscription}
+				/>
+			),
 			nextStep: StepCreateLicense.LICENSE_KEY_DETAILS,
 			stepTitle: 'Subscription',
 			title: 'Subscription',
 		},
 		[StepCreateLicense.LICENSE_KEY_DETAILS]: {
 			backStep: StepCreateLicense.SUBSCRIPTION,
-			component: <div>License</div>,
+			component: <LicenseDetails />,
 			nextStep: StepCreateLicense.SUBSCRIPTION,
 			stepTitle: 'License Key Details',
 			title: 'License Key Details',
 		},
 	};
 
+	const ProductCardInfo: ProductCardProps = {
+		licenseKeyData: {
+			endDate: 'Oct 24, 2024',
+			keyType: 'Trial',
+			startDate: 'Sep 24, 2023',
+		},
+		product: {
+			attachments: [],
+			name: {en_US: 'Test Product'},
+			productSpecifications: [],
+			skus: [
+				{
+					price: 0,
+					sku: 'TESTFREEPRODUCTSKU',
+					skuOptions: [],
+				},
+			],
+		},
+		productCreatorAccount: {
+			logoURL: undefined,
+			name: 'Test Name',
+		},
+		userAccount: {
+			emailAddress: 'test@liferay.com',
+		},
+	};
+
+	const ExtendBanner = () => (
+		<>
+			<div className="align-items-center d-flex mb-3 row">
+				<small className="col-6 col-md-4 font-weight-bold m-0">
+					Key type
+				</small>
+				<small className="col-6 col-md-4 subscription-banner-text">
+					{ProductCardInfo.licenseKeyData.keyType}
+				</small>
+			</div>
+
+			<div className="align-items-center d-flex row">
+				<small className="col-6 col-md-4 font-weight-bold m-0">
+					Start Date - Exp. Date
+				</small>
+				<small className="col-6 col-md-4 subscription-banner-text text-nowrap">
+					{ProductCardInfo.licenseKeyData.startDate} &ndash;{' '}
+					{ProductCardInfo.licenseKeyData.endDate}
+				</small>
+			</div>
+		</>
+	);
```

Comment:
Can we transform this component in an independent component?

Reply:
Hi @claricebrasil, thanks for the suggestion. The thing is: This is already a component of the Product Card.